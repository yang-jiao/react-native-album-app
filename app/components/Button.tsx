
import React from 'react'
import { Pressable, Text } from 'react-native'

import styles from './components.style'

type Props = {
    text: string
    onPress: () => void
}

const Button: React.FC<Props> = ({ text, onPress }: Props) => {
    return (
        <Pressable
                style={styles.button}
                onPress={onPress}
            >
                <Text style={styles.textStyle}>{text}</Text>
            </Pressable>

    )
}
export default Button
