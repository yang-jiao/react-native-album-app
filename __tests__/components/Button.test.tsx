import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../../app/components/Button'

describe('Button test', () => {
  const testData = {
    text: 'test',
    onPress: () => {},
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<Button text={testData.text} onPress={testData.onPress} />)
      .toJSON()
    expect(tree).toBeDefined()
    expect(tree).toMatchSnapshot()
  })
})
