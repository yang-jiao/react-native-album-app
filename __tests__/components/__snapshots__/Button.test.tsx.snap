// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2196F3",
      "borderRadius": 4,
      "elevation": 2,
      "marginTop": 15,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
